### PR TITLE
Support for request content mutation. 

### PR DESCRIPTION
**Pull request by vbothe23**
## Description of what I changed
Fixes issue: #258 

- Added an attribute `requestContent` in the `RequestMutation` class.
- In `mutateRequest()` function, added check for the request content.
       -  If the request content is modified by the plugin, then replacing the actual content with modified content.     

## E2E test

TESTED:

Please replace this with a description of how you tested your PR beyond the
automated e2e/unit tests.

## Checklist: I completed these to help reviewers :)

- [x] I have read and will follow the [review process](https://github.com/GoogleCloudPlatform/openmrs-fhir-analytics/blob/master/doc/review_process.md).
- [ ] I am familiar with Google Style Guides for the language I have coded in.

  No? Please take some time and review [Java](https://google.github.io/styleguide/javaguide.html) and [Python](https://google.github.io/styleguide/pyguide.html) style guides.

- [ ] My IDE is configured to follow the Google [**code styles**](https://google.github.io/styleguide/).

  No? Unsure? -> [configure your IDE](https://github.com/google/google-java-format).

- [ ] I have **added tests** to cover my changes. (If you refactored existing code that was well tested you do not have to add tests)
- [x] I ran `mvn clean package` right before creating this pull request and added all formatting changes to my commit.
- [ ] All new and existing **tests passed**.
- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by vbothe23** [comment]
Hi @bashir2 ,
As per our discussion, I have raised this draft PR. Please review the changes that I have implemented.

File: server/src/main/java/com/google/fhir/gateway/BearerAuthorizationInterceptor.java [hunk 1]
```diff
@@ -283,12 +283,16 @@ private void serveWellKnown(ServletRequestDetails request) {
...
-    mutation
-        .getQueryParams()
-        .forEach((key, value) -> requestDetails.addParameter(key, value.toArray(new String[0])));
+    if (!CollectionUtils.isEmpty(mutation.getQueryParams())) {
+      mutation
+          .getQueryParams()
+          .forEach((key, value) -> requestDetails.addParameter(key, value.toArray(new String[0])));
+    }
+    if (mutation.getRequestContent().length > 0) {
+      requestDetails.setRequestContents(mutation.getRequestContent());
```
Comment (bashir2) [inline server/src/main/java/com/google/fhir/gateway/BearerAuthorizationInterceptor.java:295]: Can you please add a unit-test that exercises this? For example something like [this](https://github.com/google/fhir-gateway/blob/f0788a647799d92ce8111782367281e6b859f503/server/src/test/java/com/google/fhir/gateway/BearerAuthorizationInterceptorTest.java#L275).

Also can you please fill-up the TESTED field of the PR description and describe how you have tested this change. Especially if your AccessChecker plugin is open-source, that would be great to point to that as well.

**Review by bashir2 — COMMENTED** [review]
Thanks @vbothe23 for making this change. Please see the comment below; I also have some more high-level/architecture comments which I will share on issue #258.

**Comment by vbothe23** [comment]
Thanks @bashir2, for sharing the concerns about the request mutation. As discussed we decided to proceed with the `prostProcess` approach, which mutate the response from the FHIR server, rather than the request data goes into the server. Thus our problem will be solved without these code changes.

**Pull request closed without merging**